zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/ParagraphRuntimeInfos.java

```diff
@@ -0,0 +1,29 @@
+package org.apache.zeppelin.notebook;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * 
+ *
+ */
+public class ParagraphRuntimeInfos {
+
+  String propertyName;
+  String label;
+  String group;
+  List<String> values;
+
+  public ParagraphRuntimeInfos(String propertyName, String label, String group) {
+    this.propertyName = propertyName;
+    this.label = label;
+    this.group = group;
+  }
+
+  public void addValue(String value) {
+    if (values == null) {
+      values = new ArrayList<>();
```

Comment:
Should we create values in the constructor ? Otherwise at least this method is not thread-safe (we might create 2 values in 2 threads)